### DISCUSSION TITLE
Are the audiotools libraries really supported for ESP32 S3 ?

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [pschatzmann/arduino-audio-tools#561] [Q&A] [answered]
That the question i am posing,
the sd & sdfat tests can't compile and show a bunch of errors,even for the sd & sdfat  players a2dp ,
all these examples are ok with ESP32.
I have the last audiotools update and the last es32 boards update: 2.06
To be more informative,i would like to use the WT32-SC01-PLUS witch have a ESP32 S3,that works with  LOVYAN GFX library
for tft display.

**Comment by pschatzmann** [comment, marked as answer]
You might have overlooked [this documentation](https://github.com/pschatzmann/ESP32-A2DP/wiki/Support-for-esp32-c3%3F)
I tried to compile a sketch which was using the player with I2S and the SD library and did not get any errors...

**Comment by TangerineDreamer** [comment]
You are right for the bluetooth,but for the simple test-index-sd i have these errors:

//---------------------------------------------
In file included from C:\Users\UTILIS~1\AppData\Local\Temp\arduino_modified_sketch_904011\test-index-sd.ino:7:
C:\Users\Utilisateur\Documents\Arduino\libraries\arduino-audio-tools\src/AudioLibs/SDIndex.h: In member function 'const char* audio_tools::SDIndex<SDT, FileT>::fileNamePath(FileT&)':
C:\Users\Utilisateur\Documents\Arduino\libraries\arduino-audio-tools\src/AudioLibs/SDIndex.h:256:23: error: 'start_dir' was not declared in this scope
       file_path_str = start_dir;
                       ^~~~~~~~~
C:\Users\Utilisateur\Documents\Arduino\libraries\arduino-audio-tools\src/AudioLibs/SDIndex.h:256:23: note: suggested alternative: '_stat_r'
       file_path_str = start_dir;
                       ^~~~~~~~~
                       _stat_r
//---------------------------------------------------
The sketch with my perso settings:

//----------------------------------------------------
#include "AudioTools.h"
#include "AudioLibs/AudioKit.h"
#include "FS.h"
#include "SD.h"
#include "SPI.h"
#include "AudioLibs/SDIndex.h"

void setup() {
  Serial.begin(115200);
  while(!Serial);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning);

  SPI.begin(39,38,40,41);  
  while (!SD.begin(41)) {
       Serial.println("SD.begin failed");
       delay(1000);
  }
  SDIndex<fs::SDFS,fs::File> idx{SD};
  idx.ls(Serial, "/", "mp3","*");

}

void loop() {
  delay(1000);
}
//------------------------------------------------------

By the way,happy new year :)

**Reply by pschatzmann** [reply]
Yes, the last commit has introduced an error...

**Comment by TangerineDreamer** [comment]
I believe it,it works perfect now with SD,
THANK YOU :)

**Comment by TangerineDreamer** [comment]
I would like to list only directories,not their mp3 files,but i can't find the protocole to do this,
if i do:  idx.ls(Serial, "/","","");  but i see nothing
same for :  idx.ls(Serial, "/","/","");
how can i?

**Comment by pschatzmann** [comment]
This is definitely not in the scope of this library, so you will need to implement this yourself using the standard SD API.
Google is your friend! I suggest that you also have a look at the Arduino SD examples...

**Comment by TangerineDreamer** [comment]
I'm searching a lot on the web,what i would like to do is:

1) list all the mp3 albums,then select an album,
2) then list all the songs of the album 
3) then select the song

On the net, the examples are very basic ,same for the mp3 players,the examples only play from the root
